File: src/main/java/nl/asdproject2/ClientInputThread.java
package nl.asdproject2;

import com.sun.xml.internal.ws.client.ClientTransportException;

import java.io.DataInputStream;
import java.io.IOException;
import java.io.InputStream;

public class ClientInputThread extends Thread
{
    private InputStream inFromServer;

    public ClientInputThread(InputStream inFromServer)
    {
        this.inFromServer = inFromServer;
    }

    public void run()
    {
        while (true)
        {
            DataInputStream in = new DataInputStream(inFromServer);
            try
            {
                System.out.println(in.readUTF());
            }
            catch (IOException e)
            {
                e.printStackTrace();
            }
        }
    }
}

File: src/main/java/nl/asdproject2/kotlin/ClientInputThread.kt
package nl.asdproject2.kotlin

import com.sun.xml.internal.ws.client.ClientTransportException

import java.io.DataInputStream
import java.io.IOException
import java.io.InputStream

class ClientInputThread(private val inFromServer: InputStream) : Thread() {

    override fun run() {
        while (true) {
            val `in` = DataInputStream(inFromServer)
            try {
                println(`in`.readUTF())
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
    }
}
